src/components/MultiStepIndicator/components/StepIndicator/StepIndicator.jsx

```diff
@@ -0,0 +1,66 @@
+import React, { useMemo } from "react";
+import cx from "classnames";
+import PropTypes from "prop-types";
+import Check from "../../../Icon/Icons/components/Check";
+import { MULTI_STEP_TYPES } from "../../MultiStepConstants";
+import { baseClassName, STEP_STATUSES } from "./StepIndicatorConstants";
+import "./StepIndicator.scss";
+
+const StepIndicator = ({ stepComponentClassName, stepNumber, status, titleText, subtitleText, type }) => {
+  const ariaLabel = useMemo(() => {
+    return `Step ${stepNumber}: ${status}`;
+  }, [status]);
+
+  const baseClassNameWithType = `${baseClassName}--type-${type}`;
+  const baseClassNameWithStatus = `${baseClassName}--status-${status}`;
+
+  const getClassnames = suffix => {
+    return [`${baseClassName}${suffix}`, `${baseClassNameWithType}${suffix}`, `${baseClassNameWithStatus}${suffix}`];
+  };
+
+  return (
+    <div className={cx(baseClassName, baseClassNameWithType, stepComponentClassName)} aria-label={ariaLabel}>
+      <div className={cx(...getClassnames("__number-container"))}>
+        <span className={cx(...getClassnames("__number-container__text"))}>
+          {status === StepIndicator.statuses.FULFILLED ? <Check /> : stepNumber}
+        </span>
+      </div>
+      <div className={cx(...getClassnames("__text-container"))}>
+        <div className={cx(...getClassnames("__text-container__title"))}>
+          <span className="visually-hidden">{status}</span> {/* A11y*/}
```

Comment:
amazing!!! i wanted to write this one but didn't get there 
please please export it to a reuse-able component `<HiddenText text={text}  id={can get id for aria-labeled-by}/>

Love it!!!! 🥇